tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6499Test.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Doctrine\Tests\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group DDC-6499
+ */
+class DDC6499Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+            $this->_em->getClassMetadata(DDC6499A::class),
+            $this->_em->getClassMetadata(DDC6499B::class),
+            ]
+        );
+    }
+
+    /**
+     * Test for the bug described in issue #6499.
+     */
+    public function testIssue()
+    {
+        $a = new DDC6499A();
+        $this->_em->persist($a);
+
+        $b = new DDC6499B();
+        $a->setB($b);
+        $this->_em->persist($b);
+
+        $this->_em->flush();
+
+        // Issue #6499 will result in a Integrity constraint violation before reaching this point
+        $this->assertEquals(true, true);
+    }
+}
+
+/** @Entity */
+class DDC6499A
+{
+    /**
+     * @Id()
+     * @GeneratedValue(strategy="AUTO")
+     * @Column(name="id", type="integer")
+     */
+    private $id;
+
+    /**
+     * @OneToMany(targetEntity="DDC6499B", mappedBy="a", cascade={"persist", "remove"}, orphanRemoval=true)
```

Comment:
Drop all cascade operations, orphan removals and properties/mappings that aren't strictly related to the tests